walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java

```diff
@@ -56,6 +57,150 @@ public void setUp() {
     query = new FindMeetingQuery();
   }
 
+  @Test
+  public void optionalAttendeeIsNotConsidered() {
+    // The optional attendee will not affect the outcome
+    // Required attendees: A, B
+    // Optional attendees: C
+    // Events  :       |--A--|     |--B--|
+    //           |--------------C--------------|
+    // Day     : |-----------------------------|
+    // Options : |--1--|     |--2--|     |--3--|
+
+    Collection<Event> events = Arrays.asList(
+        new Event("Event 3", TimeRange.
+            fromStartEnd(TimeRange.START_OF_DAY, TimeRange.END_OF_DAY, true), 
+            Arrays.asList(PERSON_C)),
+        new Event("Event 1", TimeRange.fromStartDuration(TIME_0800AM, DURATION_30_MINUTES),
+            Arrays.asList(PERSON_A)),
+        new Event("Event 2", TimeRange.fromStartDuration(TIME_0900AM, DURATION_30_MINUTES),
+            Arrays.asList(PERSON_B)));
+
+    MeetingRequest request =
+        new MeetingRequest(Arrays.asList(PERSON_A, PERSON_B), DURATION_30_MINUTES);
+    request.addOptionalAttendee(PERSON_C);
+
+    Collection<TimeRange> actual = query.query(events, request);
+    Collection<TimeRange> expected =
+        Arrays.asList(TimeRange.fromStartEnd(TimeRange.START_OF_DAY, TIME_0800AM, false),
+            TimeRange.fromStartEnd(TIME_0830AM, TIME_0900AM, false),
+            TimeRange.fromStartEnd(TIME_0930AM, TimeRange.END_OF_DAY, true));
+
+    Assert.assertEquals(expected, actual);
+  }
+
+  @Test
+  public void optionalAttendeeIsConsidered() {
```

Comment:
when will the optional attendee be considered?
Could you document this in FindMeetingQuery?
BTW, where do you take optional attendee into consideration in the FindMeetingQuery?

Reply:
I haven't implemented the algorithm when the optional attendees are considered. For now, I just wrote the tests and pull request to check my tests are okay. After that, I'll implement the algorithm and create another pull request.

Reply:
I think you should send the PR includes both the changes and the test. Otherwise you'll have broken tests after this PR submitted.

Reply:
Done! I've finished the algorithm.